pytket/qir/conversion/conversion.py

```diff
@@ -0,0 +1,728 @@
+                    pyqir.IntPredicate.SGT, lower_qir, self.ssa_vars[registername]
+                )
+                upper_cond = module.module.builder.icmp(
+                    pyqir.IntPredicate.SGT, self.ssa_vars[registername], upper_qir
+                )
+
+                result = module.module.builder.and_(lower_cond, upper_cond)
+
+                condition_bit_index = command.args[-1].index[0]
+                registername = command.args[-1].reg_name
+
+                self.module.builder.call(
+                    self.set_one_bit_in_reg,
+                    [
+                        self.ssa_vars[registername],
+                        pyqir.const(self.qir_int_type, condition_bit_index),
+                        result,
+                    ],
+                )
+
+            elif isinstance(op, Conditional):
+                assert op.width == 1  # only one conditional bit
+                conditional_circuit = self._rebase_op_to_gateset(
+                    op.op, command.args[op.width :]
+                )
+                condition_bit_index = command.args[0].index[0]
+                condition_name = command.args[0].reg_name
+
+                def condition_block_true() -> None:
+                    """
+                    Populate recursively the module with the contents of the conditional
+                    sub-circuit when the condition is True.
+                    """
+                    if op.value == 1:
+                        self.circuit_to_module(conditional_circuit, module)
+
+                def condition_block_false() -> None:
+                    """
+                    Populate recursively the module with the contents of the conditional
+                    sub-circuit when the condition is False.
+                    """
+                    if op.value == 0:
+                        self.circuit_to_module(conditional_circuit, module)
+
+                if condition_name in self.ssa_vars:
+
+                    ssabool = module.builder.call(
+                        self.read_bit_from_reg,
+                        [
+                            self.ssa_vars[condition_name],
+                            pyqir.const(self.qir_int_type, condition_bit_index),
+                        ],
+                    )
+
+                    module.module.builder.if_(
+                        ssabool,
+                        true=lambda: condition_block_true(),  # type: ignore
+                        false=lambda: condition_block_false(),  # type: ignore
+                    )
+
+                else:
+                    ValueError(
+                        "circuit contruction with special condition not yet supported"
+                    )
+                    # this should be an assertion when working
+
+            elif isinstance(op, WASMOp):
+                raise ValueError(
+                    "WASM not supported yet, support expected with pytket-qir==0.3.0"
```

Comment:
Let's not mention the expected version number here.

Reply:
done in @cqc-melf
[remove wasm version](https://github.com/CQCL/pytket-qir/pull/36/commits/a44a55ebd403a1c171d73f31c9dbca47be847b40)
[a44a55e](https://github.com/CQCL/pytket-qir/pull/36/commits/a44a55ebd403a1c171d73f31c9dbca47be847b40)